ios/engine/KMEI/KeymanEngine/Classes/InputViewController.swift

```diff
@@ -163,6 +165,11 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
 
     Manager.shared.setText(context)
     Manager.shared.setSelectionRange(NSRange(newRange, in: context), manually: false)
+
+    if let webView = self.kmInputView as? WKWebView {
+      webView.evaluateJavaScript("resetContext();")
```

Comment:
All JavaScript calls should be in `KeymanWebViewController` which can be accessed through `Manager.shared.keymanWeb`.

Reply:
Maybe we can hold a weak reference to `textInput` and compare before resetting the context as well in case it's only text that changed.